addons/binding/org.openhab.binding.onebusaway/src/main/java/org/openhab/binding/onebusaway/internal/handler/StopHandler.java

```diff
@@ -174,59 +174,67 @@ private StopConfiguration loadAndCheckConfiguration() {
-                copiedRouteData.putAll(key, copy);
+            boolean alreadyFetching = !fetchInProgress.compareAndSet(false, true);
+            if (alreadyFetching) {
+                return false;
             }
+            logger.debug("Fetching data for stop ID {}", config.getStopId());
+            String url = String.format("http://%s/api/where/arrivals-and-departures-for-stop/%s.json?key=%s",
+                    apiHandler.getApiServer(), config.getStopId(), apiHandler.getApiKey());
+            URI uri;
+            try {
+                uri = new URI(url);
+            } catch (URISyntaxException e) {
+                logger.error("Unable to parse '%s' as a URI.", url);
+                updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.OFFLINE.CONFIGURATION_ERROR,
+                        "stopId or apiKey is set to a bogus value");
+                return false;
+            }
+            ContentResponse response;
+            try {
+                response = httpClient.newRequest(uri).send();
+            } catch (InterruptedException | TimeoutException | ExecutionException e) {
+                updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.OFFLINE.COMMUNICATION_ERROR, e.getMessage());
+                return false;
+            }
+            if (response.getStatus() != HttpStatus.OK_200) {
+                updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.OFFLINE.COMMUNICATION_ERROR,
+                        String.format("While fetching stop data: %d: %s", response.getStatus(), response.getReason()));
+                return false;
+            }
+            ObaStopArrivalResponse data = gson.fromJson(response.getContentAsString(), ObaStopArrivalResponse.class);
+            routeDataLastUpdateMs = data.currentTime;
+            updateStatus(ThingStatus.ONLINE);
+
+            ArrayListMultimap<String, ObaStopArrivalResponse.ArrivalAndDeparture> copiedRouteData = ArrayListMultimap
+                    .create();
+            synchronized (routeData) {
+                routeData = ArrayListMultimap.create();
+                for (ObaStopArrivalResponse.ArrivalAndDeparture d : data.data.entry.arrivalsAndDepartures) {
+                    routeData.put(d.routeId, d);
+                }
+                for (String key : routeData.keySet()) {
+                    List<ObaStopArrivalResponse.ArrivalAndDeparture> copy = Lists.newArrayList(routeData.get(key));
+                    Collections.sort(copy);
+                    copiedRouteData.putAll(key, copy);
+                }
+            }
+            for (RouteDataListener listener : routeDataListeners) {
+                listener.onNewRouteData(routeDataLastUpdateMs, copiedRouteData.get(listener.getRouteId()));
+            }
+            return true;
+        } catch (Exception e) {
+            logger.warn("Exception refreshing route data", e);
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.valueOf("Internal error"));
```

Comment:
Please pass a proper ThingStatusDetail, see other examples and please pass the message while updating the status.

And given the message is in the status you can change the logger to debug.

Reply:
Let me try adding the ThingStatusDetail. Thank you for your help; I'm fairly new to OH2.

Reply:
It should look something like:

```java
updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.OFFLINE.COMMUNICATION_ERROR, e.getMessage());
```